sale_timesheet_task_change_project/models/account_analytic_line.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2022 Coop IT Easy SCRLfs
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class AccountAnalyticLine(models.Model):
+
+    _inherit = "account.analytic.line"
+
+    @api.multi
+    def write(self, vals):
+        result = True
+        for rec in self:
+            rec_vals = vals.copy()
+            self._adjust_so_line_link_write_vals(rec, rec_vals)
+            # it must be an explicit super(), because it must be run on rec, not self.
+            result = super(AccountAnalyticLine, rec).write(rec_vals) and result
```

Comment:
I don't understand the logic of setting result to True, and adding and setting `result = super(AccountAnalyticLine, rec).write(rec_vals) and result`. What's the idea ?

Reply:
it seems that `.write()` returns `True` when the operation succeeds. i guess if it fails it does not return and raises an exception instead. since here we’re iterating over the records, each `.write()` call will return something, and all of them must be `True` to return `True`. i’m not sure it is really needed. i also don’t like the explicit `super()` call with arguments. maybe there is a cleaner way to do this.